### PR TITLE
Typer fixes

### PR DESCRIPTION
**Pull request by Blaisorblade**
WIP fixing #5008 and #5010.

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/typer/Typer.scala [hunk 1]
```diff
@@ -1529,6 +1529,7 @@ class Typer extends Namer
         ctx.error(i"$psym is extended twice", tree.pos)
       seenParents += psym
       if (tree.isType) {
+        checkSimpleKinded(result) // Not needed for constructor calls, as type arguments will be inferred.
```
Comment (hrhino) [inline compiler/src/dotty/tools/dotc/typer/Typer.scala]: will they be inferred? I'm unconvinced that

```scala
class Siz[T](i: Int)
class Zle extends Siz(1)
```

and 

```scala
class Siz[T]
class Zle extends Siz
```

should be treated differently w/r/t superclass targs.

**Comment by Blaisorblade** [comment]
(Since GitHub hides @hrhino's comment):
> will they be inferred? I'm unconvinced that
> ```scala
> class Siz[T](i: Int)
> class Zle extends Siz(1)
> ```
> and 
> ```scala
> class Siz[T]
> class Zle extends Siz
> ```
> should be treated differently w/r/t superclass targs.

Well `T` in `Siz(1)` will be inferred as in a normal method call (by interpolation).

If we want uniformity, we could also infer arguments to *all* parents, but that seems more error-prone, no?

**Comment by allanrenucci** [comment]
@Blaisorblade Can you split the fix for enum into another PR?

**Comment by Blaisorblade** [comment]
So the failure in http://dotty-ci.epfl.ch/lampepfl/dotty/7075/5 is because master is still broken by the Java 9 PR.

**Comment by Blaisorblade** [comment]
Split into #5027, #5028, #5029, #5030.

**Pull request closed without merging**